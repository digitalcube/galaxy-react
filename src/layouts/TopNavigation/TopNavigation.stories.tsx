import React from 'react';
import { Meta, Story } from '@storybook/react';
import { TopNavigation, TopNavigationProps } from './TopNavigation'

const meta: Meta = {
  title: 'Layouts/TopNavigation',
  component: TopNavigation,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<TopNavigationProps> = args => <TopNavigation {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
    serviceName: 'Shifter',
    today: new Date('2010-01-01')
}