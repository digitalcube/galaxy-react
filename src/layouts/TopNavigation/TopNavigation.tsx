import React, {FC} from 'react'
import { FaUser, FaSignOutAlt } from 'react-icons/fa'
import { SearchForm } from '../../components/SearchForm/SearchForm'
import { NavLinkItem, Nav, NavbarBrand, NavbarDropdown, NavbarDropdownItem } from '../../routings'

export type TopNavigationProps = {}


export const TopNavigation: FC<TopNavigationProps> = () => {
    return (
        <nav className="navbar navbar-expand-lg">
            <NavbarBrand href="/" name="shifter" />
            <SearchForm onChange={() => undefined} onSubmit={e => e.preventDefault()}/>
            <Nav>
                    <NavLinkItem active to="#">Sites</NavLinkItem>
                    <NavLinkItem to="#">Teams</NavLinkItem>
                    <NavLinkItem to="#">Guides</NavLinkItem>
                    <NavbarDropdown
                        avatar={{
                            size: 32,
                            email: "test@example.com",
                            rounded: true
                        }}
                    >
                        <NavbarDropdownItem href="#">
                            Account<FaUser />
                        </NavbarDropdownItem>
                        <NavbarDropdownItem href="#">
                            Log out <FaSignOutAlt />
                        </NavbarDropdownItem>
                    </NavbarDropdown>
                </Nav>
        </nav>
    )
}