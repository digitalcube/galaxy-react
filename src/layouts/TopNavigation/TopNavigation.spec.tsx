import React from 'react';
import { render } from '@testing-library/react'
import { Default as TopNavigation } from './TopNavigation.stories';

describe('TopNavigation', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(<TopNavigation {...{
        ...TopNavigation.args,
    }}/>);
    expect(asFragment()).toMatchSnapshot()
  });
});
