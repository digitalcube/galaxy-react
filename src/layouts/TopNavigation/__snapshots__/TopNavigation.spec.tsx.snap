// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavigation to match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <img
        alt="Shifter"
        height="32"
        width="24"
      />
    </a>
    <form
      class="form-inline my-2 my-lg-0 flex-grow-1 d-flex position-relative form-search"
    >
      <div
        class="input-group"
      >
        <svg
          class="fas fa-search form-search-icon position-absolute"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
        <input
          aria-label="Search"
          class="form-control"
          placeholder="search"
          type="text"
          value=""
        />
      </div>
    </form>
    <button
      aria-label="Toggle navigation"
      class="navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse flex-grow-0 collapse"
    >
      <ul
        class="ml-auto navbar-nav nav"
      >
        <li
          class="nav-item active"
        >
          <a
            class="rounded font-weight-bold nav-link"
          >
            Sites
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
          >
            Teams
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
          >
            Guides
          </a>
        </li>
        <li
          class="dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="nav-link font-weight-bold dropdown-toggle nav-link"
            data-toggle="dropdown"
            href="#"
            role="button"
          >
            <img
              alt="Gravatar for test@example.com"
              class="react-gravatar p-0  rounded-circle"
              height="32"
              src="//www.gravatar.com/avatar/55502f40dc8b7c769880b10874abc9d0?d=retro&r=g&s=32"
              srcset="//www.gravatar.com/avatar/55502f40dc8b7c769880b10874abc9d0?d=retro&r=g&s=64 2x"
              width="32"
            />
          </a>
          <div
            aria-hidden="true"
            class="dropdown-menu dropdown-menu-right"
            role="menu"
            tabindex="-1"
          >
            <button
              class="dropdown-item"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <a
                class="dropdown-item px-3 py-2 d-flex align-items-center justify-content-between"
                href="#"
              >
                Account
                <i
                  class="fas fa-user"
                />
              </a>
            </button>
            <button
              class="dropdown-item"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              <a
                class="dropdown-item px-3 py-2 d-flex align-items-center justify-content-between"
                href="#"
              >
                Log Out
                <i
                  class="fas fa-sign-out-alt"
                />
              </a>
            </button>
          </div>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
